OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldaccess/reflection/ReflectionRelatedFieldAccessesAnalysis.scala

```diff
@@ -0,0 +1,1055 @@
+                addFieldAccess(state.callContext, data.pc, _ => data.receiver, _ => data.param, allMatchers)
+            }
+
+        AllocationsUtil.continuationForAllocation[NameDepender, ContextType](
+            eps, state.callContext, data => (data.nameVar, data.callerStatements),
+            _.isInstanceOf[NameDepender],
+            data => failure(data.pc, data.receiver, data.param, data.matchers)
+        ) { (data, _, allocationIndex, stmts) =>
+                val name = StringUtil.getString(allocationIndex, stmts)
+
+                val nameMatcher = retrieveSuitableMatcher[Set[String]](
+                    name.map(Set(_)),
+                    data.pc,
+                    v => new NameBasedFieldMatcher(v)
+                )
+
+                if (nameMatcher ne NoFieldsMatcher) {
+                    val matchers = data.matchers + nameMatcher
+                    val allMatchers = matchers +
+                        MatcherUtil.retrieveClassBasedFieldMatcher(
+                            data.callerContext,
+                            data.classVar,
+                            ClassDepender(data.pc, data.receiver, data.param, matchers, data.classVar, data.callerStatements),
+                            data.pc,
+                            stmts,
+                            project,
+                            () => failure(data.pc, data.receiver, data.param, matchers),
+                            onlyFieldsExactlyInClass = !data.matchers.contains(PublicFieldMatcher)
+                        )
+
+                    addFieldAccess(state.callContext, data.pc, _ => data.receiver, _ => data.param, allMatchers)
+                }
+            }
+
+        AllocationsUtil.continuationForAllocation[ClassDepender, ContextType](
+            eps, state.callContext, data => (data.classVar, data.callerStatements),
+            _.isInstanceOf[ClassDepender], data => failure(data.pc, data.receiver, data.param, data.matchers)
+        ) { (data, allocationContext, allocationIndex, stmts) =>
+                val classes = TypesUtil.getPossibleClasses(
+                    allocationContext, allocationIndex, data,
+                    stmts, () => failure(data.pc, data.receiver, data.param, data.matchers),
+                    onlyObjectTypes = false
+                )
+
+                val matchers = data.matchers +
+                    retrieveSuitableMatcher[Set[ObjectType]](
+                        Some(classes.map {
+                            tpe => if (tpe.isObjectType) tpe.asObjectType else ObjectType.Object
+                        }),
+                        data.pc,
+                        v => new ClassBasedFieldMatcher(v, !data.matchers.contains(PublicFieldMatcher))
+                    )
+
+                addFieldAccess(state.callContext, data.pc, _ => data.receiver, _ => data.param, matchers)
+            }
+
+        AllocationsUtil.continuationForAllocation[(ClassDepender, V, Array[Stmt[V]]), ContextType](
```

Comment:
What is this continuation for? It looks broken, since the given type is a 3-tuple, but the instanceOf check checks for a 2-tuple.

Reply:
This continuation was for "for name classes", which got refactored in the meantime. I will (once again) copy them from `ReflectionRelatedCalls`.

Reply:
As discussed in person, we will migrate this to the `ClassForName` property as happened in reflective calls already.